src/configuration-requestor.ts

```diff
@@ -1,132 +1,48 @@
 import { IHttpClient } from './http-client';
-import {
-  ConfigStoreHydrationPacket,
-  IConfiguration,
-  StoreBackedConfiguration,
-} from './i-configuration';
-import { BanditVariation, BanditParameters, Flag, BanditReference } from './interfaces';
+
+export type ConfigurationRequestorOptions = {
+  wantsBandits?: boolean;
+};
 
 // Requests AND stores flag configurations
 export default class ConfigurationRequestor {
-  private banditModelVersions: string[] = [];
-  private readonly configuration: StoreBackedConfiguration;
+  private readonly options: ConfigurationRequestorOptions;
 
   constructor(
     private readonly httpClient: IHttpClient,
-    private readonly flagConfigurationStore: IConfigurationStore<Flag>,
-    private readonly banditVariationConfigurationStore: IConfigurationStore<
-      BanditVariation[]
-    > | null,
-    private readonly banditModelConfigurationStore: IConfigurationStore<BanditParameters> | null,
+    private readonly configurationStore: ConfigurationStore,
+    options: Partial<ConfigurationRequestorOptions> = {},
   ) {
-    this.configuration = new StoreBackedConfiguration(
-      this.flagConfigurationStore,
-      this.banditVariationConfigurationStore,
-      this.banditModelConfigurationStore,
-    );
-  }
-
-  public isFlagConfigExpired(): Promise<boolean> {
-    return this.flagConfigurationStore.isExpired();
-  }
-
-  public getConfiguration(): IConfiguration {
-    return this.configuration;
+    this.options = {
+      wantsBandits: true,
+      ...options,
+    };
   }
 
-  async fetchAndStoreConfigurations(): Promise<void> {
+  async fetchConfiguration(): Promise<Configuration | null> {
     const configResponse = await this.httpClient.getUniversalFlagConfiguration();
-    if (!configResponse?.flags) {
-      return;
-    }
-
-    const flagResponsePacket: ConfigStoreHydrationPacket<Flag> = {
-      entries: configResponse.flags,
-      environment: configResponse.environment,
-      createdAt: configResponse.createdAt,
-      format: configResponse.format,
-    };
-
-    let banditVariationPacket: ConfigStoreHydrationPacket<BanditVariation[]> | undefined;
-    let banditModelPacket: ConfigStoreHydrationPacket<BanditParameters> | undefined;
-    const flagsHaveBandits = Object.keys(configResponse.banditReferences ?? {}).length > 0;
-    const banditStoresProvided = Boolean(
-      this.banditVariationConfigurationStore && this.banditModelConfigurationStore,
-    );
-    if (flagsHaveBandits && banditStoresProvided) {
-      // Map bandit flag associations by flag key for quick lookup (instead of bandit key as provided by the UFC)
-      const banditVariations = this.indexBanditVariationsByFlagKey(configResponse.banditReferences);
-
-      banditVariationPacket = {
-        entries: banditVariations,
-        environment: configResponse.environment,
-        createdAt: configResponse.createdAt,
-        format: configResponse.format,
-      };
-
-      if (
```

Comment:
We check to see if the set of bandit models we've already loaded satisfies the models referenced by the latest flag config. Limits calls to fetch the bandits to once daily (bandit model recomputation cycle)

Reply:
oh, good catch